src/uniq.js

```diff
@@ -18,4 +20,39 @@ var uniqWith = require('./uniqWith');
  *      R.uniq([1, '1']);     //=> [1, '1']
  *      R.uniq([[42], [42]]); //=> [[42]]
  */
-module.exports = uniqWith(equals);
+module.exports = _curry1(function uniq(list) {
+  /* global Set */
+  var item, set, idx, len, items, uniqs;
+
+  /**
+   * Uses a native `Set` instance where possible for
+   * removing duplicate items. Items that implement
+   * the fantasy-land Setoid spec fallback to using
+   * `_contains` to support custom equality behaviour.
+   */
+  if (typeof Set === 'function') {
```

Comment:
The behaviour is intentional, though there may be a better approach. The reason we disallow top-level assignments (aside from the module imports) is that they could conflict. If two modules were both to define a top-level `x` variable, we could end up with the following output:

```javascript
var x = ...;  // from src/f.js

var f = function(...) { ... };

var x = ...;  // from src/g.js

var g = function(...) { ... };
```

We could wrap each module in an IIFE in __dist/ramda.js__, but at least at the time I wrote the build script the readability of the generated file was considered important.